improver/utilities/cube_manipulation.py

```diff
@@ -60,6 +64,22 @@ def collapsed(cube, *args, **kwargs):
     original_methods = cube.cell_methods
     new_cube = cube.collapsed(*args, **kwargs)
     new_cube.cell_methods = original_methods
+
+    # demote escalated datatypes as required
+    if new_cube.dtype in FLOAT_TYPES:
+        new_cube.data = new_cube.data.astype(FLOAT_DTYPE)
+
+    collapse_coords = find_dimension_coordinate_mismatch(cube, new_cube)
```

Comment:
`collapsed_coords` would be clearer - the collapsing has already happened by this point.

Wouldn't it be better to take this value from `args[0]` instead?

Reply:
Almost certainly.  I couldn't figure out the best way - but args[0] looks correct, as it'll respond correctly to a single coordinate or list of coordinates.